### ISSUE TITLE
Tags- Product is found after search with tag when the language selected is not correct

### ISSUE THREAD

**Issue by marwachelly** [PrestaShop/PrestaShop#28294]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Product is found after search with tag when the language selected is not correct

https://user-images.githubusercontent.com/89604350/164190027-dcfc6edc-947e-4e8c-9e5a-358d2b18fc8c.mp4

### Expected behavior

Product should be found with tags only when the correct language is selected in FO

### Steps to reproduce
1- Frech install
2- add the languages : български език (Bulgarian) and Azərbaycan dili (Azerbaijani)
3- Go to a product V2 
4- add the tag "MƏŞHUR MƏHSULLAR" ( contains characters from Azerbaijani language) when english language is selected and save
5- search the tag from FO when български език (Bulgarian) or Azərbaycan dili (Azerbaijani) is selected => The product is find with the tag
6- delete the tag from product page => the tag is deleted and the product is still found whith the tag

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

8.0

**Comment by zuk3975** [comment]
Hi, is it a regression of product page migration? Have you found the same issue in 177?

**Comment by zuk3975** [comment]
WIP.
I've been debugging this for a while, so just some notes about the issue (for myself and/or someone else if needed).
1. The search results already changes when we ~~add additional language~~ {**modified**  `apparently its not even related to languages. I just needed to perform the search couple times and eventually it doesn't find anything`}, without touching tags. (e.g. In freshly installed shop with default language EN, if I search "MƏŞHUR MƏHSULLAR" I can see 5 products (customizable mug being the first one found. While debugging I found out all of them are found by fuzzy search by phrase "mug". That pharse is returned by levenshtein algorithm). Once I add additional language and search for same phrase again, there are no products found in neither languages. If I remove that language, there are no products found in the EN language anymore.
2. Described issue with the tags is also related to fuzzy search (main search doesn't find any products as expected, but the fuzzy search does). It seems after adding/removing tags, something changes/caches up (couldn't yet find what) and after that, fuzzy search starts finding the product by the tag which doesn't actually exist anymore.

**Comment by zuk3975** [comment]
Ok im lost with this (and other search related issues) - I installed 2 **identical shops** and **right after fresh install** I searched for phrase "MƏŞHUR MƏHSULLAR". Both shops **produced different results**. One found 5 mugs, other found nothing. Im stuck with this for 2 days already and I can only say these are not related 
to product page v2 and not specifically with tags, but rather the fuzzy search and search indexing.
![Screenshot from 2022-09-06 14-55-21](https://user-images.githubusercontent.com/31609858/188630290-5b4082e2-cce5-42a9-82be-5830c7ecb4e5.png)
![Screenshot from 2022-09-06 14-55-13](https://user-images.githubusercontent.com/31609858/188630296-2e7ca9e7-614a-49a5-b16c-be820bd563c5.png).

One quick fix at least seems to be to re-index all products in `Shop parameters -> Search` when unexpected search results are produced.
Disabling fuzzy search in `Shop paramers -> Search` also has impact. Seems that feature is responsible for issues when products are found by tags by which they shouldn't be found.
